### PR TITLE
Fix return code if folder is missing

### PR DESCRIPTION
**Pull request by schiessle**
fix https://github.com/nextcloud/end_to_end_encryption/issues/40

throw a 409 (conflict) if the parent folder doesn't exists.

### CODE REVIEW

File: lib/Connector/Sabre/LockPlugin.php [hunk 1]
```diff
@@ -157,18 +153,24 @@ protected function checkUserAgent($userAgent, $path) {
...
+		$parent = dirname($path);
+		$parent = ($parent === '.') ? '/' : $parent;
+		if ($this->server->tree->nodeExists($parent)) {
+			return $this->server->tree->getNodeForPath($parent);
 		}
 
-		return $node;
+		// If neither the actual node, nor the parent exists we throw a exception.
+		// According to the WebDAV specification it should result in 409 (conflict)
+		throw new Conflict();
```
Comment (rullzer) [inline lib/Connector/Sabre/LockPlugin.php]: What if I do a profind to `<server>/foo/bar/baz/x.txt` and that whole structure doesn't exist? Then I should get back a 404.

This probably has to be extended by having multiple https://github.com/nextcloud/end_to_end_encryption/blob/7f06848679ea59e0b5c5414edfe2d20dcc8b0ae2/lib/Connector/Sabre/LockPlugin.php#L99 so you know when you are trying to create a folder etc and thus when to throw the 409 etc.

Or you catch it in https://github.com/nextcloud/end_to_end_encryption/blob/7f06848679ea59e0b5c5414edfe2d20dcc8b0ae2/lib/Connector/Sabre/LockPlugin.php#L110 and just return so the other plugins handle it?

Comment (tobiasKaminsky) [reply lib/Connector/Sabre/LockPlugin.php]: Yes, currently a head to a non-existing path throws 409 instead of 404.

**Comment by schiessle** [comment]
@rullzer can you have another look. now it should work as expected

**Comment by tobiasKaminsky** [comment]
👍 working fine for me

**Pull request merged**